### PR TITLE
Add CI Mix Packages

### PR DESCRIPTION
**Pull request by KaylaBrady**
Ticket here: https://app.asana.com/0/1199987702860136/1200240419411843

Adds static code analysis packages - credo & dialyzer. Got a clean output running dialyzer, credo pointed out one missing moduledoc. I added a simple one based on the phoenix docs [here](https://hexdocs.pm/phoenix/telemetry.html#the-telemetry-supervisor). This was an auto-generated file -- is this actively used for metrics in other projects, or is this file unnecessary?

### CODE REVIEW

**Comment by paulswartz** [comment]
@KaylaBrady 
> Adds static code analysis packages - credo & dialyzer. Got a clean output running dialyzer, credo pointed out one missing moduledoc. I added a simple one based on the phoenix docs [here](https://hexdocs.pm/phoenix/telemetry.html#the-telemetry-supervisor). This was an auto-generated file -- is this actively used for metrics in other projects, or is this file unnecessary?

Telemetry only came out in 2018 after our larger projects were already started, so I don't believe any of them use it. That doesn't meant we shouldn't start using it for new projects, though!

**Comment by KaylaBrady** [comment]
@paulswartz 
> 
> Telemetry only came out in 2018 after our larger projects were already started, so I don't believe any of them use it. That doesn't meant we shouldn't start using it for new projects, though!

Is there a different way that older projects tracked these metrics ad-hoc, or do they not have these more granular metrics? Interested for my own learning.

**Comment by paulswartz** [comment]
@KaylaBrady 
> @paulswartz
> 
> > Telemetry only came out in 2018 after our larger projects were already started, so I don't believe any of them use it. That doesn't meant we shouldn't start using it for new projects, though!
> 
> Is there a different way that older projects tracked these metrics ad-hoc, or do they not have these more granular metrics? Interested for my own learning.

More ad-hoc. For example, the API has a wrapper around `:timer.tc/1`: https://github.com/mbta/api/blob/HEAD/apps/state/lib/logger.ex

File: test/support/conn_case.ex [hunk 1]
```diff
@@ -32,10 +32,11 @@ defmodule DraftWeb.ConnCase do
   end
 
   setup tags do
-    :ok = Ecto.Adapters.SQL.Sandbox.checkout(Draft.Repo)
+    alias Ecto.Adapters.SQL.Sandbox
```
Comment (KaylaBrady) [inline test/support/conn_case.ex:35]: aliasing nested module based on credo output

File: README.md [hunk 2]
```diff
@@ -11,3 +11,9 @@ To start your Phoenix server:
 
 Now you can visit [`localhost:4000`](http://localhost:4000) from your browser.
 
+
+Useful when developing:
+
+  * Test elixir code with `mix test --coverage`
+  * Format elixir code with `mix format`
+  * Lint elixir code with `mix dialyzer` and `mix credo --strict`
```
Comment (KaylaBrady) [inline README.md:19]: specified `--strict` since that is what will be used by the Elixir CI action.

**Review by paulswartz — APPROVED** [review]
🍰

**Pull request merged**